### ISSUE TITLE
Download and upload on every new instance of VS Code with no settings changed

### ISSUE THREAD

**Issue by agross** [shanalikhan/code-settings-sync#402]
**Visual Studio Code Version :** 1.16.0
**Code Settings Sync Version :** 2.8.3
**Operating System :** Windows 10 and macOS
**Occurs While:** Opening VS Code
**Proxy Enabled:** No
**Gist Id:** https://gist.github.com/agross/c6a31b72704b476730646876caaa7140/revisions

Hi,

I have a Mac and a Windows 10 machine, both fully patched and running latest.

I noticed that immediately after opening a new instance of VS Code on both machines VS Code Settings Sync first downloads the Gist and a couple of seconds later uploads the Gist again. I have configured to display the sync summary, and this is where the first problem occurs: As both the download and the upload try to write that file, the upload will result in a conflict that is displayed as a notification in VS Code. Perhaps the plugin should write to a different file on each download or upload.

As you can see from the Gist revision log, nothing changed between Gist versions except for `lastUpload` and `lastDownload`.

On Windows, I have this in `syncLocalSettings.json` (macOS has no special settings):

```json
{ "replaceCodeSettings": {"git.path":"C:\\Tools\\Git for Windows\\bin\\git.exe"} }
```

It doesn't matter whether I opened VS Code on the other machine in the meantime.

**Comment by shanalikhan** [comment]
Yes it is happening, can do write the complete steps for reproducing at my end in detail
https://github.com/shanalikhan/code-settings-sync/issues/402
https://github.com/shanalikhan/code-settings-sync/issues/396

**Comment by agross** [comment]
Did you fix it?

**Comment by shanalikhan** [comment]
No its duplicate of issue, #396 

> As both the download and the upload try to write that file, the upload will result in a conflict that is displayed as a notification in VS Code

VS Code will show file conflicts when the file is open in code and extension try to rewrite the complete file.
What you can do it to close the file which extension will be going to write such as `settings.json` when the upload / download process is in progress.

**Comment by agross** [comment]
It's not the settings.json that is affected, it's the syncSummary.txt. Most of often download and upload happen in rapid succession which leave me with a 2 second time window tomorrow close the summary, which I often miss. I'd rather see a timestamped version of the summary rather than forcing your users to be close a file on every start of VS Code.

But I digress. The main issue I was reporting is that the sync happens for no reason (e. g. no settings changed). It seems that the change detection is somehow broken. I run Code on three different machines and the amount of time I need fixing sync issues is nontrivial, up to the point that I now have a git clone of my GitHub Gist to verify what my actual plugin list is when VS Code Settings Sync once again removed extensions or downgraded extensions when I haven't been using Code for a while on one of the machines.

To summarize, I don't feel this is a duplicate, except for the SyncSummary.txt part.

**Comment by shanalikhan** [comment]
@arnohovhannisyan

This is one of the scenario, regarding autoUpload.

Can you check this on new version branch. ?

**Comment by agross** [comment]
> Can you check this on new version branch. ?

Unfortunately, I can't. I've migrated away to Syncing.

**Comment by arnohovhannisyan** [comment]
> Can you check this on new version branch. ?

This issue seems to be resolved on v3.2.8

**Comment by shanalikhan** [comment]
> Unfortunately, I can't. I've migrated away to Syncing.

😞 

Closing this issue. Let me know if there is any problem.
